### PR TITLE
Outline matches

### PR DESCRIPTION
**Pull request by bomsy**
Associated Issue: #800

### Summary of Changes

* Added new search overlay for outlining matches
* Added css for styling outlines

### Testing

* [x] passes `npm test`
* [x] passes `npm run lint`

### Screenshots/Videos 
outlines

### CODE REVIEW

**Comment by bomsy** [comment]
Was not sure about the theme color variable to use for selections.

File: public/js/utils/source-search.js [hunk 1]
```diff
@@ -32,6 +33,31 @@ function getSearchCursor(cm, query, pos) {
     typeof query == "string" && query == query.toLowerCase());
 }
 
+function searchOverlay(query) {
+  query = query.replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, "\\$&");
```
Comment (jasonLaster) [inline public/js/utils/source-search.js]: lol. this PR is crazy town, do you have some context on how you came up with it :)

File: public/js/lib/codemirror-mozilla.css [hunk 2]
```diff
@@ -125,6 +125,14 @@ selector in floating-scrollbar-light.css across all platforms. */
   background-repeat: repeat-x;
 }
 
+.cm-selecting {
+  background: none;
+  border-width: 1px;
```
Comment (jasonLaster) [inline public/js/lib/codemirror-mozilla.css]: this is a minor thing, but would you mind trying outline as well. I notice that border pushes the text, which is a slight visual tick that atom and i believe other editors avoid

File: public/js/utils/source-search.js (hunk 1 above)
Comment (bomsy) [inline public/js/utils/source-search.js]: Had a look into an example of how code mirror does highlighting of text. Basically Codemirror's `addOverlay` function takes a mode object. The mode object should have a token function which is applied as each token in the file is parsed. https://codemirror.net/doc/manual.html#option_mode

The regex just escapes  any special characters in the input

Comment (bomsy) [inline public/js/utils/source-search.js]: cool .. will do!

**Comment by bomsy** [comment]
Thanks!
We could definately replace it.

**Comment by jasonLaster** [comment]
Awesome. Let's open a new PR to make it consistent. Also less scary

**Comment by bomsy** [comment]
ok :)

**Pull request merged**